runtime/lua/vim/lsp.lua

```diff
@@ -477,6 +489,18 @@ function lsp.start_client(config)
   return client.id, nil
 end
 
+--- @deprecated
+--- Starts and initializes a client with the given configuration.
+--- @param config vim.lsp.ClientConfig Configuration for the server.
+--- @return integer? client_id |vim.lsp.get_client_by_id()| Note: client may not be
+---         fully initialized. Use `on_init` to do any actions once
+---         the client has been initialized.
+--- @return string? # Error message, if any
+function lsp.start_client(config)
+  vim.deprecate('vim.lsp.start_client()', 'vim.lsp.start()', '0.12')
```

Comment:
```suggestion
  vim.deprecate('vim.lsp.start_client()', 'vim.lsp.start()', '0.13')
```

So that users have the full 0.11 - 0.12 cycle to switch to `lsp.start`, otherwise they'd need to add compat logic and check `vim.fn.has(nvim-0.11)`

Reply:
And just to note these numbers aren't strict versions in which things will be removed. Just a minimum version.